src/server/scripts/World/npcs_special.cpp

```diff
@@ -2017,6 +2016,7 @@ class npc_experience : public CreatureScript
         ClearGossipMenuFor(player);
         bool noXPGain = player->HasFlag(PLAYER_FLAGS, PLAYER_FLAGS_NO_XP_GAIN);
         bool doSwitch = false;
+        auto toggleXpCost = sWorld->getIntConfig(CONFIG_TOGGLE_XP_COST);
```

Comment:
Do you need to change to```int32```

Reply:
with `auto` the compiler will automatically figure out the type according to the return type of `getIntConfig`